arbnode/resourcemanager/resource_management.go

```diff
@@ -31,37 +32,72 @@ var (
 // prior to RPC request handling.
 //
 // Must be run before the go-ethereum stack is set up (ethereum/go-ethereum/node.New).
-func Init(conf *Config) {
-	if conf.MemLimitPercent > 0 {
-		node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
-			var c limitChecker
-			c, err := newCgroupsMemoryLimitCheckerIfSupported(conf)
-			if errors.Is(err, errNotSupported) {
-				log.Error("No method for determining memory usage and limits was discovered, disabled memory limit RPC throttling")
-				c = &trivialLimitChecker{}
-			}
+func Init(conf *Config) error {
+	if conf.MemFreeLimit == "" {
+		return nil
+	}
+
+	limit, err := parseMemLimit(conf.MemFreeLimit)
+	if err != nil {
+		return err
+	}
 
-			return newHttpServer(srv, c), nil
+	node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
+		var c limitChecker
+		c, err := newCgroupsMemoryLimitCheckerIfSupported(limit)
+		if errors.Is(err, errNotSupported) {
+			log.Error("No method for determining memory usage and limits was discovered, disabled memory limit RPC throttling")
+			c = &trivialLimitChecker{}
 		}
+
+		return newHttpServer(srv, c), nil
 	}
+	return nil
+}
+
+func parseMemLimit(limitStr string) (int, error) {
+	var (
+		limit int = 1
+		s     string
+	)
+	_, err := fmt.Sscanf(limitStr, "%d%s", &limit, &s)
+	if err != nil {
```

Comment:
nit: you can inline this as well:

if _, err := ...; err != nil {
}

Reply:
Thanks, will fix these in the next PR.